### PR TITLE
Feature tag

### PR DESCRIPTION
**Pull request by Theomorais**

### CODE REVIEW

**Comment by GauthierPLM** [comment]
Instead of using "molecule-dev-environment", it would be better to use "molecule" as preference' namespace.
For some redondent part of the code, it would be better to use functions to shorten the code and avoid repetition (in particular for plugin loading functions).

File: lib/molecule-dev-environment.js [hunk 1]
```diff
@@ -46,22 +101,169 @@ export default {
...
-    renderPinnedPlans();
+    if (
+      global.atom.config.get("molecule-dev-environment.pinnedPlans") === true
+    ) {
+      renderDevtoolBar();
+      renderPinnedPlans();
+    }
     renderControlPanel();
 
+    if (
```
Comment (PoignardAzur) [inline lib/molecule-dev-environment.js]: All these `if config.get(...) == null config.set(...)` can be removed.

File: lib/molecule-dev-environment.js [hunk 2]
```diff
@@ -98,6 +300,45 @@ export default {
             ),
           );
         },
+        "molecule:free-terminal-toggle": () => {
```
Comment (PoignardAzur) [inline lib/molecule-dev-environment.js]: All these commands can be removed. Settings can and should be changed directly from the settings view in Atom.

File: lib/molecule-dev-environment.js [hunk 3]
```diff
@@ -112,10 +353,194 @@ export default {
...
+      const linter = registerIndie({
+        name: "molecule",
+      });
+      this.subscriptions.add(linter);
+      store.subscribe(handleStateChange(linter));
+      store.dispatch(featureLoad("Linter"));
+    }
+  },
+
+  toggleFreeTerminal() {
```
Comment (PoignardAzur) [inline lib/molecule-dev-environment.js]: All the toggle methods can be removed. Same reasoning.

File: menus/molecule-dev-environment.json [hunk 4]
```diff
@@ -0,0 +1,71 @@
+{
```
Comment (PoignardAzur) [inline menus/molecule-dev-environment.json]: This file can be removed. Same reasoning.

File: lib/molecule-dev-environment.js [hunk 5]
```diff
@@ -46,22 +101,169 @@ export default {
...
-      "Flowtype.js",
-      "Gulp.js",
-      "Jest.js",
-      "Npm.js",
-      "ReactNative.js",
-      "Shell.js",
-      "Testcafe.js",
-    ].forEach(tool => provideDevtool(store, DevToolsControllerInstance)(tool));
+
+    if (
```
Comment (PoignardAzur) [inline lib/molecule-dev-environment.js]: All these ifs should be condensed into one loop. Something like:

``` js
for (const pluginName of Object.keys(pluginRoots)) {
    if (global.atom.config.get(`molecule.${pluginName}`)) {
      pluginsTable.push(pluginRoots[pluginName]);
      store.dispatch(featureLoad(pluginName));
    }
}
```

File: lib/ExecutionControlEpic/TaggingFeature/Actions/FeatureLoad.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+"use babel";
```
Comment (PoignardAzur) [inline lib/ExecutionControlEpic/TaggingFeature/Actions/FeatureLoad.js]: I'm not sure this action is very useful, especially if it's only dispatched at startup, especially especially if it's only used for logging. If think it should be removed to reduce bloat.

File: lib/molecule-dev-environment.js (hunk 5 above)
Comment (alanzanattadev) [reply lib/molecule-dev-environment.js]: I especially think that the table creation should be done with map / filter and the dispatch in a for / forEach, separation of concerns, two separated block of codes

**Pull request closed without merging**